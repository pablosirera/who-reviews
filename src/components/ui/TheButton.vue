<template>
  <button
    @click="$emit('click')"
    :class="buttonClasses"
    data-testid="the-button"
    class="mt-5 p-2 border-solid rounded border-black bg-blue-500 text-white font-bold focus:outline-none hover:opacity-75"
  >
    {{ text }}
  </button>
</template>

<script>
export default {
  name: 'TheButton',
  props: {
    text: {
      type: String,
      required: true,
    },
    size: {
      type: String,
      default: 'medium',
    },
  },
  computed: {
    buttonClasses() {
      return {
        'w-16': this.size === 'small',
        'w-24': this.size === 'medium',
        'w-40': this.size === 'large',
      }
    },
  },
}
</script>
