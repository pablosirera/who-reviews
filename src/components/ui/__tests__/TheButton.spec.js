import { shallowMount } from '@vue/test-utils'
import TheButton from '@/components/ui/TheButton'

describe('The Button', () => {
  const defaultProps = { text: 'hello' }
  const build = (propsData = defaultProps) => {
    const wrapper = shallowMount(TheButton, {
      propsData,
    })

    return {
      wrapper,
    }
  }

  it('should match snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should set class w-16 when is small size', () => {
    const { wrapper } = build({ text: 'hello', size: 'small' })
    wrapper.find('[data-testid="the-button"]')
    expect(wrapper.classes()).toContain('w-16')
  })

  it('should set class w-24 when is small size', () => {
    const { wrapper } = build({ text: 'hello', size: 'medium' })
    wrapper.find('[data-testid="the-button"]')
    expect(wrapper.classes()).toContain('w-24')
  })

  it('should set class w-40 when is small size', () => {
    const { wrapper } = build({ text: 'hello', size: 'large' })
    wrapper.find('[data-testid="the-button"]')
    expect(wrapper.classes()).toContain('w-40')
  })
})
