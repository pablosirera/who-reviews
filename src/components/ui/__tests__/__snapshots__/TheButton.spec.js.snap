// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button should match snapshot 1`] = `
<button
  class="mt-5 p-2 border-solid rounded border-black bg-blue-500 text-white font-bold focus:outline-none hover:opacity-75 w-24"
  data-testid="the-button"
>
  
  hello

</button>
`;
